.gitlab/catalyst/pipeline.yml

```diff
@@ -0,0 +1,256 @@
+    - export SLURM_JOB_ID=$(squeue -h -n "${JOB_NAME}" -o "%A")
+    - cd ci_test/integration_tests
+    - python3 -m pytest -s -vv --durations=0 --junitxml=results.xml
+  artifacts:
+    when: always
+    paths:
+      - ci_test/integration_tests/results.xml
+    reports:
+      junit: ci_test/integration_tests/results.xml
+
+# This is a dummy job that checks the output of the Catch2
+# testing. This is one of the gotchas of GitLab: The Catch2 test
+# executables are artifacts of the build procedure, but we don't save
+# the entire build as artifacts (it would be very large and there's no
+# need since the rest of the tests run on the installed targets (for
+# better or worse)). For maximum clarity from the web interface, we
+# write a file when the Catch tests fail but then return success from
+# the script that runs them. This means that the build job will fail
+# iff the build fails. This job will be marked as "failed" if any of
+# the catch tests failed, and the pytest-based unit/integration tests
+# will run any time the build succeeds.
+check catch2 tests:
+  extends:
+    - .catalyst common
+  stage: test
+  dependencies:
+    - build and install
+  script:
+    - ([[ $(find ${RESULTS_DIR} -name "catch-tests-failed.txt" | wc -l) -eq 0 ]])
+  artifacts:
+    reports:
+      junit: ${RESULTS_DIR}/*.xml
+
+# The spack environment that we use is tied to the pipeline. Thus it
+# is meaningless after the pipeline dies, but it still needs to be
+# cleaned up. It is always run. The shell script verifies the
+# environment existence before trying to delete it. An environment
+# would not exist if the pipeline failed before a certain point in the
+# build stage (namely, when the build script creates the environment).
+remove spack environment:
+  extends: .catalyst common
+  stage: deallocate
+  variables:
+    GIT_STRATEGY: none
+  when: always
+  script:
+    - echo "== CLEANING UP SPACK RESOURCES =="
+    - source ${HOME}/spack_repos/spack_${SYSTEM_NAME}.git/share/spack/setup-env.sh
+    - export SPACK_ARCH_TARGET=$(spack arch -t)
+    - export FULL_ENV_NAME=lbann-${SPACK_ENV_NAME}-${SPACK_ARCH_TARGET}
+    - |
+      if [[ -n "$(spack env list | grep -e ${FULL_ENV_NAME})" ]] ;
+      then
+        spack env rm --yes-to-all ${FULL_ENV_NAME}
+      fi
+
+# This frees the allocation we obtained in "allocate lc resources".
+release allocation:
+  stage: deallocate
+  extends: .catalyst common
+  variables:
+    GIT_STRATEGY: none
+  when: always
+  script:
+    - echo "== RELEASING RESOURCES =="
+    - export JOB_ID=$(squeue -h -n "${JOB_NAME}" -o "%A")
+    - ([[ -n "${JOB_ID}" ]] && scancel ${JOB_ID})
+
+# This loads the spack shell integration and then loads the
+# environment. Since this relies on the environment for this pipeline
+# existing, it should only be used after the build stage (i.e., in
+# testing).
+.uses spack environment:
+  before_script:
+    - source ${HOME}/spack_repos/spack_catalyst.git/share/spack/setup-env.sh
```

Comment:
To make this easier to copy and paste, we should probable replace catalyst here with ${SYSTEM_NAME}